### PR TITLE
Count Anagrams in JavaScript

### PR DESCRIPTION
**Pull request by Coder-Srinivas**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Given two strings word and text, return the number of 
substrings where text contains any anagram of word.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue: #2304

### CODE REVIEW

**Comment by Coder-Srinivas** [comment]
@siddharth25pandey @HarshCasper I have added the code for count anagrams in JavaScript. You can test it [here](https://binarysearch.com/problems/Anagram-Substrings). Please let me know if any changes are required. 😊

File: JavaScript/cp/CountAnagrams.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            /* Removing the first character of the window */
+            alpha[text.charCodeAt(window.shift()) - 97]++;
+        }
+    }
+
+    return ans;
+}
+
+/* Printing the number of anagrams of the word "for" in the text "trfororfideriorofrufrx" */
+console.log(countAnagrams("for", "trfororfideriorofrufrx"))
```
Comment (HarshCasper) [inline JavaScript/cp/CountAnagrams.js]: Make the input dynamic

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Closing this PR due to inactivity.

**Comment by Coder-Srinivas** [comment]
@HarshCasper I have added dynamic input using the readline() module. Please let me know if any changes are required.

File: JavaScript/cp/CountAnagrams.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+1
+
+Example 2:
+
+Input:
+abc
+bcabxabc
+
+Output:
+3
```
Comment (ankitaggarwal23) [inline JavaScript/cp/CountAnagrams.js]: Write a proper display message how you are reading and printing output.
Remove the extra testcase one is enough.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Coder-Srinivas** [comment]
@ankitaggarwal23 Made the changes requested.

File: JavaScript/cp/CountAnagrams.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+/* 
+Example 1:
+
+Input:
+ab
+aab
+
+Output:
+1
```
Comment (ankitaggarwal23) [inline JavaScript/cp/CountAnagrams.js]: Write the I/p and O/p the way your code is producing

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by Coder-Srinivas** [comment]
@ankitaggarwal23 Made the changes requested.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**